frame/benchmarking/src/analysis.rs

```diff
@@ -415,6 +503,53 @@ mod tests {
 		}
 	}
 
+	#[test]
+	fn test_linear_regression() {
+		let ys = vec![
+			3797981.0,
+			37857779.0,
+			70569402.0,
+			104004114.0,
+			137233924.0,
+			169826237.0,
+			203521133.0,
+			237552333.0,
+			271082065.0,
+			305554637.0,
+			335218347.0,
+			371759065.0,
+			405086197.0,
+			438353555.0,
+			472891417.0,
+			505339532.0,
+			527784778.0,
+			562590596.0,
+			635291991.0,
+			673027090.0,
+			708119408.0,
+		];
+		let xs = vec![
+			0.0, 1.0, 2.0, 3.0, 4.0, 5.0, 6.0, 7.0, 8.0, 9.0, 10.0, 11.0, 12.0, 13.0, 14.0, 15.0,
+			16.0, 17.0, 18.0, 19.0, 20.0,
+		];
+
+		let (intercept, params, errors) =
+			raw_linear_regression(xs.clone(), ys.clone(), 1, true).unwrap();
+		assert_eq!(intercept as i64, -2712997);
```

Comment:
I deliberately left it in to make sure the base linear regression routine works correctly (e.g. you can plug in those numbers into an alternative implementation, and get the same numbers). Numerical algorithms are a pain to debug since you can never tell on a first glance whether they're actually correct, so I think having this sort of a sanity check is a good thing.